consensus/hotstuff/votecollector/combined_vote_processor.go

```diff
@@ -0,0 +1,163 @@
+package votecollector
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/model/flow"
+	msig "github.com/onflow/flow-go/module/signature"
+)
+
+// CombinedVoteProcessorFactory generates CombinedVoteProcessor instances
+func CombinedVoteProcessorFactory(log zerolog.Logger, proposal *model.Proposal) (*CombinedVoteProcessor, error) {
+	processor := &CombinedVoteProcessor{
+		log:   log,
+		block: proposal.Block,
+	}
+	processor.done.Store(false)
```

Comment:
is this call necessary?

Reply:
Note that `processor.done` is _not_ a pointer. Instead, to increase data locality, I included it directly. 
But I guess we could still do: 
```golang
	processor := &CombinedVoteProcessor{
		log:   log,
		block: proposal.Block,
		done: *atomic.NewBool(false),
	}
```
(?)